Core/include/Acts/Utilities/Any.hpp

```diff
@@ -0,0 +1,427 @@
+    if (m_handler == nullptr) {
+      return;
+    }
+
+    void* to = m_data.data();
+    const void* from = fromAny.m_data.data();
+    if (m_handler->typeSize > SIZE) {
+      // stored on heap: interpret buffer as pointer
+      to = *reinterpret_cast<void**>(m_data.data());
+      from = *reinterpret_cast<void* const*>(fromAny.m_data.data());
+    }
+
+    if (m_handler->copyConstruct == nullptr) {
+      // trivially copyable
+      m_data = fromAny.m_data;
+    } else {
+      m_handler->copyConstruct(from, to);
+    }
+  }
+
+  void copy(const AnyBase& fromAny) {
+    if (m_handler == nullptr) {
+      return;
+    }
+
+    void* to = m_data.data();
+    const void* from = fromAny.m_data.data();
+    if (m_handler->typeSize > SIZE) {
+      // stored on heap: interpret buffer as pointer
+      to = *reinterpret_cast<void**>(m_data.data());
+      from = *reinterpret_cast<void* const*>(fromAny.m_data.data());
+    }
+
+    if (m_handler->copy == nullptr) {
+      // trivially copyable
+      m_data = fromAny.m_data;
+    } else {
+      m_handler->copy(from, to);
+    }
+  }
+
+  template <typename T>
+  static void destroyImpl(void* ptr) {
+    assert(ptr != nullptr && "Address to destroy is nullptr");
+    T* obj = static_cast<T*>(ptr);
+    if constexpr (sizeof(T) <= SIZE) {
+      // stored in place: just call the destructor
+      _ACTS_ANY_DEBUG("Destroy local at: " << ptr);
+      obj->~T();
+    } else {
+      // stored on heap: delete
+      _ACTS_ANY_DEBUG("Delete heap at: " << obj);
+      delete obj;
+    }
+  }
+
+  template <typename T>
+  static void moveConstructImpl(void* from, void* to) {
+    _ACTS_ANY_DEBUG("move const: " << from << " -> " << to);
+    assert(from != nullptr && "Source is null");
+    assert(to != nullptr && "Target is null");
+    T* _from = static_cast<T*>(from);
+    /*T* ptr =*/new (to) T(std::move(*_from));
+  }
+
+  template <typename T>
+  static void moveImpl(void* from, void* to) {
+    _ACTS_ANY_DEBUG("move: " << from << " -> " << to);
+    assert(from != nullptr && "Source is null");
+    assert(to != nullptr && "Target is null");
+
+    T* _from = static_cast<T*>(from);
+    T* _to = static_cast<T*>(to);
+
+    (*_to) = std::move(*_from);
+  }
+
+  template <typename T>
+  static void copyConstructImpl(const void* from, void* to) {
+    _ACTS_ANY_DEBUG("copy const: " << from << " -> " << to);
+    assert(from != nullptr && "Source is null");
+    assert(to != nullptr && "Target is null");
+    const T* _from = static_cast<const T*>(from);
+    /*T* ptr =*/new (to) T(*_from);
+  }
+
+  template <typename T>
+  static void copyImpl(const void* from, void* to) {
+    _ACTS_ANY_DEBUG("copy: " << from << " -> " << to);
+    assert(from != nullptr && "Source is null");
+    assert(to != nullptr && "Target is null");
+
+    const T* _from = static_cast<const T*>(from);
+    T* _to = static_cast<T*>(to);
+
+    (*_to) = *_from;
+  }
+
+  alignas(std::max_align_t) std::array<char, SIZE> m_data{};
```

Comment:
Maybe use `std::byte` instead of `char`?

Reply:
Can do, but shouldn't make a difference, no?

Reply:
Main major difference is losing the "`char*` is interpreted as a string by `std::cout`" footgun, which is nice when debug-printing around.

Reply:
Ok